lib/ide/src/view/text_editor.rs

```diff
@@ -45,54 +88,65 @@ impl TextEditor {
         let text_size    = 16.0;
         let properties   = TextFieldProperties {font,text_size,base_color,size};
         let text_field   = TextField::new(&world,properties);
+        let key_listener = None;
+        let controller_clone = controller.clone_ref();
+        let text_field_clone = text_field.clone();
+        executor::global::spawn(async move {
+            if let Ok(content) = controller_clone.read_content().await {
+                text_field_clone.write(&content);
+            }
+        });
         world.add_child(&text_field);
 
-        Self {text_field,padding,position,size}.initialize()
+        let data = TextEditorData {controller,text_field,padding,position,size,key_listener};
+        let rc = Rc::new(RefCell::new(data));
+        Self{rc}.initialize()
     }
 
     fn initialize(self) -> Self {
+        let text_editor = Rc::downgrade(&self.rc);
+        let closure     = move |event:KeyboardEvent| {
+            const S_KEY : u32 = 83;
+            if event.ctrl_key() && event.key_code() == S_KEY {
+                if let Some(text_editor) = text_editor.upgrade() {
+                    text_editor.borrow().save();
+                }
+            }
+        };
+        let closure : Box<dyn FnMut(KeyboardEvent)> = Box::new(closure);
+        let callback                                = Closure::wrap(closure);
+        let key_listener = KeyboardListener::new("keydown".into(), callback);
```

Comment:
Literal here. Either replace it with enum or consider making a function like `KeyboardListener::new_key_down` so KeyboardListener does not force its users to use literal on every instantiation.